### PR TITLE
Change mean to logsumexp in GW and NICER likelihoods

### PR DESCRIPTION
**Pull request by ThibeauWouters**
As pointed out by @tsunhopang , the use of mean over the individual log-likelihoods computed in the GW and NICER likelihoods was mathematically incorrect (oops). This PR fixes that to use logsumexp instead and make the maths work out again. 

The inference results have negligible impact, though. Interestingly, the inference is slightly faster now, likely since the penalty from violating the TOV constraint in the GW likelihood now has less weight compared to the mean. 

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved numerical stability of likelihoods for gravitational-wave and neutron-star data by replacing averaged log-probability aggregation with more robust log-sum-exp aggregation.

* **Bug Fixes / Reliability**
  * Improved sampler initialization and internal state handling to stabilize sampling behavior.

* **Chores**
  * Tooling: updated recorded Python version to 3.12 and migrated CI tool invocations to a UV-managed workflow.

* **Documentation**
  * Minor label cleanup in contributor-facing documentation.

### CODE REVIEW

File: jesterTOV/inference/likelihoods/nicer.py [hunk 1]
```diff
@@ -239,8 +240,8 @@ def process_sample_maryland(mass: Float) -> Float:
         )
 
         # Average over all samples for each group
-        logL_amsterdam = jnp.mean(amsterdam_logprobs)
-        logL_maryland = jnp.mean(maryland_logprobs)
+        logL_amsterdam = logsumexp(amsterdam_logprobs)
+        logL_maryland = logsumexp(maryland_logprobs)
 
         # Average the two groups (equal weights)
         log_likelihood = (logL_amsterdam + logL_maryland) / 2.0
```
[the remark this answers is not in the document]
Comment (ThibeauWouters) [reply jesterTOV/inference/likelihoods/nicer.py]: Dang it, thanks. Should not hastily submit PRs before leaving the office it seems :)
